pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -86,12 +86,12 @@ func NewAllocatorCIDRRange(cidr *net.IPNet, allocatorFactory allocator.Allocator
 	base := utilnet.BigForIP(cidr.IP)
 	rangeSpec := cidr.String()
 
-	if utilnet.IsIPv6CIDR(cidr) {
-		// Limit the max size, since the allocator keeps a bitmap of that size.
-		if max > 65536 {
-			max = 65536
-		}
-	} else {
+	// Limit the max size, since the allocator keeps a bitmap of that size in etcd.
+	// Kubernetes Scalability only tests O(10K) services
+	// https://github.com/kubernetes/kubernetes/issues/94029
+	if max > 65536 {
+		max = 65536
+	} else if !utilnet.IsIPv6CIDR(cidr) {
```

Comment:
this shouldn't be else, right?

Reply:
We only suppress the broadcast address if it is within the range.
If we need to limit the max, means that the broadcast address is out of the range, since it is the last address of the range.